### PR TITLE
PD-302: Port <Portal> to TypeScript

### PR DESCRIPTION
**Pull request by jetpacmonkey**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [PD-302](https://jira.mongodb.org/browse/PD-302)

## 🛠 Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: packages/Portal/src/Portal.spec.tsx [hunk 1]
```diff
@@ -14,12 +14,37 @@ describe('packages/Portal', () => {
...
+
+    return el;
+  }
+
   test(`appends portal content to document body`, () => {
-    expect(document.body.firstChild.outerHTML).toBe(
+    const firstChild = document.body.firstChild;
+    const lastChild = document.body.lastChild;
+
+    if (firstChild == null || !isElement(firstChild)) {
```
Comment (hswolff) [inline packages/Portal/src/Portal.spec.tsx]: Ditto other PR - why throws and not assertions?

File: packages/Portal/src/Portal.spec.tsx [hunk 2]
```diff
@@ -14,12 +14,37 @@ describe('packages/Portal', () => {
...
+    if (el == null) {
+      throw new Error(`Could not find element: ${id}`);
+    }
+
+    return el;
+  }
+
   test(`appends portal content to document body`, () => {
-    expect(document.body.firstChild.outerHTML).toBe(
+    const firstChild = document.body.firstChild;
```
Comment (hswolff) [inline packages/Portal/src/Portal.spec.tsx]:
```suggestion
    const { firstChild, lastChild } = document.body;
```

😱

Comment (jetpacmonkey) [reply packages/Portal/src/Portal.spec.tsx]: Yeesss

File: packages/Portal/src/Portal.spec.tsx (hunk 1 above)
Comment (jetpacmonkey) [reply packages/Portal/src/Portal.spec.tsx]: Ditto other PR response: TypeScript type narrowing

File: packages/Portal/src/Portal.spec.tsx [hunk 3]
```diff
@@ -14,12 +14,36 @@ describe('packages/Portal', () => {
     </div>,
   );
 
+  function isElement(el: Node): el is HTMLElement {
```
Comment (DesignerDave) [inline packages/Portal/src/Portal.spec.tsx]: Will be nice once we move this to `lib`.

Comment (jetpacmonkey) [reply packages/Portal/src/Portal.spec.tsx]: I guess I should make the same `| null` improvement here that I did in the other branch. Definitely looking forward to having a shared util :)

Comment (DesignerDave) [reply packages/Portal/src/Portal.spec.tsx]: Indeed

**Pull request merged**